daemon/udp_listener.c

```diff
@@ -47,6 +50,13 @@ static void udp_listener_incoming(int fd, void *p, uintptr_t x) {
 			return;
 		}
 
+		if (is_compressed(buf)) {
+			ret = uncompress_data(buf, len, buf_tmp, 0x10000);
```

Comment:
This irks me for two reasons.

1) Using a single leading lower case `x` to detect compressed data seems very unreliable, and
2) The `udp_listener` facility is not exclusive to the NG protocol support.

Those two together make this approach look very prone to false positives. Instead I would suggest to either use a stronger magic cookie to mark compressed data, OR move the support for decompression into `control_ng`. Or maybe both :smile: